.github/workflows/pr-matrix.yaml

```diff
@@ -18,8 +18,23 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        SKU: [windows-intel]
-        TestTarget: [check-hlsl-d3d12, check-hlsl-warp-d3d12, check-hlsl-vk, check-hlsl-clang-d3d12, check-hlsl-clang-warp-d3d12, check-hlsl-clang-vk]
+        SKU: [windows-amd, windows-intel, windows-nvidia]
+        TestTarget: [check-hlsl-d3d12, check-hlsl-vk, check-hlsl-clang-d3d12, check-hlsl-clang-vk]
+
+    uses: ./.github/workflows/build-and-test-callable.yaml
+    with:
+      OS: windows
+      SKU: ${{ matrix.SKU }}
+      TestTarget: ${{ matrix.TestTarget }}
+      OffloadTest-branch: ${{ github.event.pull_request.head.sha }}
+      LLVM-ExtraCMakeArgs: -DCMAKE_C_COMPILER=clang-cl -DCMAKE_CXX_COMPILER=clang-cl -DOFFLOADTEST_USE_CLANG_TIDY=On
+
+  Exec-Tests-Windows-Warp:
+    strategy:
+      fail-fast: false
+      matrix:
+        SKU: [windows-amd]
```

Comment:
I'm not sure how to pick a SKU randomly or based on availability, so I picked the AMD machine because it has the better CPU and the most RAM to work with between the NVIDIA and AMD machines. 

I'm not sure yet what the specs of the Intel machine are, nor if we would would want to keep running the Warp tests on it because it's not a lab machine.

Reply:
This seems reasonable. The Intel machine I have is old and slow, so doing less on it is probably good.